rsqueakvm/interpreter.py

```diff
@@ -304,27 +318,60 @@ def getreceiverclass(self, s_context):
         return s_context.w_receiver().safe_getclass(self.space)
 
     def getblockmethod(self, s_context):
-        return s_context.blockmethod
+        return s_context.blockmethod()
 
     def loop_bytecodes(self, s_context, may_context_switch=True):
         old_pc = 0
+        jump_back_pc = 0
+        unrollings = 0
         if not jit.we_are_jitted() and may_context_switch:
             self.quick_check_for_interrupt(s_context)
         method = s_context.w_method()
+        frame_size = method.frame_size()
         while True:
             pc = s_context.pc()
             if pc < old_pc:
-                if jit.we_are_jitted():
-                    # Do the interrupt-check at the end of a loop, don't interrupt loops midway.
-                    self.jitted_check_for_interrupt(s_context)
-                self.jit_driver.can_enter_jit(
-                    pc=pc, self=self, method=method,
-                    w_class=self.getreceiverclass(s_context),
-                    blockmethod=self.getblockmethod(s_context),
-                    s_context=s_context)
+                if jump_back_pc == old_pc:
```

Comment:
This allows us to manually unroll a variable amount of times. At least one time is useful in Squeak, because of the way loops are compiled by the Squeak compiler (see also the jittests for unrolling)